test/test_lava_test_plans.py

```diff
@@ -0,0 +1,115 @@
+from lava_test_plans.__main__ import main
+
+from unittest import TestCase
+import sys
+import glob
+import os
+from multiprocessing import Pool
+
+
+def call_lava_test_plan_testcases(param):
+    variable_input_file, device, testcase = param
+    sys.argv = ["lava_test_plans", "--dry-run", "--variables", variable_input_file, "--device-type", device, "--test-case", testcase]
+    return main() == 0
+
+
+def call_lava_test_plan_testplans(param):
+    variable_input_file, device, testplan = param
+    sys.argv = ["lava_test_plans", "--dry-run", "--variables", variable_input_file, "--device-type", device, "--test-plan", testplan, "--test-lava-validity"]
+    return main() == 0
+
+
+def call_lava_test_plan_testcases_project(param):
+    variable_input_file, device, testcase, project_device_path = param
+    sys.argv = ["lava_test_plans", "--dry-run", "--variables", variable_input_file, "--testplan-device-path", project_device_path, "--device-type", device, "--test-case", testcase, "--test-lava-validity"]
+    return main() == 0
+
+
+def call_lava_test_plan_testplans_project(param):
+    variable_input_file, device, testplan, project_device_path = param
+    sys.argv = ["lava_test_plans", "--dry-run", "--variables", variable_input_file, "--testplan-device-path", project_device_path, "--device-type", device, "--test-plan", testplan, "--test-lava-validity"]
+    return main() == 0
+
+
+class TestLavaTestPlans(TestCase):
+
+    def test_testcases_without_project(self):
+        devices = [os.path.basename(d) for d in glob.glob("lava_test_plans/devices/*")]
+        testcases = [os.path.basename(d) for d in glob.glob("lava_test_plans/testcases/*.yaml")]
+        variable_input_file = "variables.ini"
+        with Pool() as p:
+            for device in devices:
+                tests = [(variable_input_file, device, testcase) for testcase in testcases]
+                results = p.map(call_lava_test_plan_testcases, tests)
+                self.assertTrue(all(results))
```

Comment:
concurrency under a single test case is probably not a good idea. if more than one device would fail, the test will fail with a different failed assertion everytime it is run. if you want to not duplicate the test code, you can use a pameterized test using `pytest`'s features, like this:

```python
import pytest

@pytest.mark.parametrize("x", range(10))
def test_multiple_items(x):
    assert x >= 0
```
When running, you get something like this:

```
$ pytest -v
[...]
collected 10 items                                                                                       

tests/test_foo.py::test_multiple_items[0] PASSED                                                   [ 10%]
tests/test_foo.py::test_multiple_items[1] PASSED                                                   [ 20%]
tests/test_foo.py::test_multiple_items[2] PASSED                                                   [ 30%]
tests/test_foo.py::test_multiple_items[3] PASSED                                                   [ 40%]
tests/test_foo.py::test_multiple_items[4] PASSED                                                   [ 50%]
tests/test_foo.py::test_multiple_items[5] PASSED                                                   [ 60%]
tests/test_foo.py::test_multiple_items[6] PASSED                                                   [ 70%]
tests/test_foo.py::test_multiple_items[7] PASSED                                                   [ 80%]
tests/test_foo.py::test_multiple_items[8] PASSED                                                   [ 90%]
tests/test_foo.py::test_multiple_items[9] PASSED                                                   [100%]
[...]
```

Then, after you have this working, you can use a pytest plugin to run tests in parallel, like [pytest-parallel](https://pypi.org/project/pytest-parallel/) or other.

Reply:
fixed